pkg/controller/installation/installation_controller.go

```diff
@@ -445,6 +480,240 @@ func isUpgradeReconcile(installation *integreatlyv1alpha1.RHMI) bool {
 	return false
 }
 
+func allProductsReconciled(installation *integreatlyv1alpha1.RHMI) bool {
+	stages := installation.Status.Stages
+
+	return (verifyClusterRHSSOVersions(stages) &&
```

Comment:
I wonder if these functions could be generic, e.g. `verifyProductVersion(stages, stage, product, operatorVersion, productVersion)`

Reply:
this is done in another PR. This could should no longer exist. I wonder if this is another rebasing error @CathalOConnorRH ?